toolkit/tools/pkg/licensecheck/testdata/licensetestchecker.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Tool to validate test data for the licensecheck package unit tests.
+
+package main
+
+import (
+	"os"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/exe"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/pkg/licensecheck"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app           = kingpin.New("licensetestchecker", "Checks test data for licenses.")
+	licenses      = app.Flag("licenses", "Path to the input file of license file paths to check for false negatives.").Required().ExistingFile()
+	licensesOut   = app.Flag("licenses-output", "Path to the output file to list all false negatives.").Required().String()
+	docs          = app.Flag("docs", "Path to the input file of doc file paths to check for false positives.").Required().ExistingFile()
+	docsOut       = app.Flag("docs-output", "Path to the output file to list all false positives.").Required().String()
+	otherFiles    = app.Flag("other-files", "Path to the input file of other file paths to check for false positives.").Required().ExistingFile()
+	otherFilesOut = app.Flag("other-files-output", "Path to the output file to list all false positives.").Required().String()
+	exceptionFile = app.Flag("exception-file", "Path to the file containing the list of exceptions to the license check.").Required().ExistingFile()
+)
+
+func main() {
+	const delimiter = "`"
+
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	logger.InitStderrLog()
+
+	exceptions, err := licensecheck.LoadLicenseExceptions(*exceptionFile)
+	// Check if there was an error loading the license exceptions
+	if err != nil {
+		logger.Log.Fatalf("Failed to load license exceptions: %v", err)
+	}
+
+	realLicenseLines, err := file.ReadLines(*licenses)
+	if err != nil {
+		logger.Log.Fatalf("failed to read input file: %v", err)
+	}
+
+	// Each line will be of the form <pkg>`/usr/share/licenses/<pkg>/<subpath>
+	filesNotDetectedAsLicense := []string{}
```

Comment:
It might be good to split this function up a little to make it easier to read.

Reply:
Refactored the data gen stuff. Did this as well as moved to .json format.